include/accessor.h

```diff
@@ -261,44 +286,81 @@ class accessor<DataT, Dims, Mode, target::device> : public detail::accessor_base
 	}
 
 	template <typename Functor>
-	accessor(const ctor_internal_tag /* tag */, const buffer<DataT, Dims>& buff, handler& cgh, Functor rmfn) {
+	accessor(const ctor_internal_tag /* tag */, const buffer<DataT, Dims>& buff, handler& cgh, const Functor& rmfn) {}
+
+	size_t get_linear_offset(const id<Dims>& index) const { return detail::get_linear_index(m_buffer_range, index - m_index_offset); }
+};
+
+template <typename DataT, access_mode Mode>
+class accessor<DataT, 0, Mode, target::device> : public detail::accessor_base<DataT, 0, Mode, target::device> {
```

Comment:
Not sure what your motivations were for doing this as a template specialization, and I'm also not sure how I feel about it. Maybe we can have a discussion about this. Some thoughts:

- It avoids having a ton of SFINAE everywhere, which is obviously nice. OTOH I think it might actually be useful to have the overloads receiving a range mapper be enabled, and then throw a static assertion to guide users towards the correct overload. Maybe we'd even want to accept it as long as the `rmfn` is an `all` range mapper? I'm thinking of generic programming contexts here.
- It saves space, at least two byte, by not including the range/offset, which is important for device accessors at least. Maybe we could achieve the same with some conditional inheritance hackery, though.
- It duplicates the multi-pass (or soon, hydration) logic, which is not great.

Reply:
This was primarily motivated by the unnecessary data members, as you already recognized, and the fact that `operator[]` exists only for `Dims > 0` while `operator*` is defined only if `Dims == 0`. Other than constructors there is very little overlap between the two cases currently, but in case we decide to implement `operator[](id<0>)` as an extension (which I'm very much in favor of), this distinction becomes less clear.

Reply:
I'm in favour of having less code duplication in this case (and offering `operator[](id<0>)`).

Reply:
Implemented as such.